libtcmu_config.c

```diff
@@ -164,18 +167,36 @@ struct tcmu_config *tcmu_config_new(void)
 	return cfg;
 }
 
-void tcmu_config_destroy(struct tcmu_config *cfg)
+static void cancel_config_thread(struct tcmu_config *cfg)
 {
-	struct tcmu_conf_option *opt;
+	pthread_t thread_id = cfg->thread_id;
+	void *join_retval;
+	int ret;
 
-	darray_foreach(opt, tcmu_options) {
-		if (opt->type == TCMU_OPT_STR)
-			free(opt->opt_str);
+	ret = pthread_cancel(thread_id);
+	if (ret) {
+		tcmu_err("pthread_cancel failed with value %d\n", ret);
+		return;
 	}
 
-	darray_free(tcmu_options);
+	pthread_join(thread_id, &join_retval);
+	if (ret) {
+		tcmu_err("pthread_join failed with value %d\n", ret);
+		return;
+	}
 
-	free(cfg);
+	if (join_retval != PTHREAD_CANCELED)
+		tcmu_err("unexpected join retval: %p\n", join_retval);
+}
+
+void tcmu_cancel_config_thread(struct tcmu_config *cfg)
+{
+	cancel_config_thread(cfg);
+}
+
+void tcmu_config_destroy(struct tcmu_config *cfg)
+{
+	cancel_config_thread(cfg);
 }
```

Comment:
It does not matter to me. Unless you have a specific use case that needs it one way or another, I would say do what is easier.